gordo_components/dataset/base.py

```diff
@@ -111,11 +122,21 @@ def join_timeseries(
             logging.debug(series.head(3))
             logging.debug(series.tail(3))
 
-            resampled = series.resample(resolution, label="left").mean()
+            resampled = series.resample(resolution, label="left").agg(
+                aggregation_methods
+            )
+            # If several aggregation methods are provided, agg gives the column names
+            # like "max" and "mean", so we have to append the original series names to
+            # the column.
+            if isinstance(resampled, pd.DataFrame):
+                resampled.columns = [
+                    f"{series.name}_{agg_col_name}"
```

Comment:
Now I am confused, what happens if you have a single column? Do you not get the aggregation method as the column name? Will the default (old, with mean) path not change the column names?